### ISSUE TITLE
Support colouring nicks by mode (@+) and possibly hostmask.

### ISSUE THREAD

**Issue by Zarthus** [weechat/weechat#168]
Hi, I've used mIRC for years and one feature I really liked is colouring names by mode. This adds consistency and gives a visual effect that makes it really easy to determine who is what.

I've added images just so it is clear how it works in mIRC. While I don't expect the same setup to be copied, I think the extra context is nice.

[How my setup works in mIRC](http://i.imgur.com/U3WcYvY.png), and [how it looks in production](http://i.imgur.com/ADYlgZL.png)

This colours both the names in the nicklist, and in general chat.

It could be as simple as 

```
/set weechat.color.use_modecolor "yes"
/set weechat.color.ops "lightmagenta"
/set weechat.color.voices "green"
/set weechat.color.none "gray"
```

I personally am not that fond of the random colours.

**Comment by Shawn-Smith** [comment]
>If this is implemented please add options for users with a "~", "!", "%" or "&" prefix as well. Basically an option for owner, protected ops, ops, hops, voice.

I think this should be added in such a way that the user can specify arbitrary prefixes and assign colors to them.

Something like: `weechat.color = ~&@:red,%:green,+:blue`
Would make ~, &, and @ red, % green, and + blue

**Comment by Mikaela** [comment]
I think that if this is implemented, this should listen what the server says to support. At freenoode `PREFIX=(ov)@+`.

You see PREFIX & others when you connect to server and usually also with `/version`.

**Comment by Shawn-Smith** [comment]
>I think that if this is implemented, this should listen what the server says to support. At freenoode PREFIX=(ov)@+.

I didn't mean to add prefixes to servers that don't support them, just the ability to color random prefixes for the servers that do use them.

>Something like: `weechat.color = ~&@:red,%:green,+:blue`

Would still only color @ and + on freenode, for example.

**Comment by BlacklightShining** [comment]
I would absolutely _love_ coloring by hostname. Been wanting that since the beginning; it's about the only thing my previous client has over WeeChat. :)

As for coloring by nick prefix, don't forget to include a default. And maybe it should go by modeletter instead of prefix, to be consistent with `irc.color.nick_prefixes`.

**Comment by Shawn-Smith** [comment]
I don't care if it goes by modeletter, as long as it only colors the modeletter if it is in the `PREFIX=` token, otherwise you'll get +q as owner/founder on some ircds, and +q as quiet/mute on others.

**Comment by somenet** [comment]
+1 on this one. This is basically the show-stopper for me to switch from irssi.

I modified my nicklist.pl (on irssi) to not only colorize nicks based on the modes dynamically retrieved on connecting, but to also sort opers as if they had at least a configurable mode set in the channel (default: +h) and also color them them (default: red).

![screen](https://cloud.githubusercontent.com/assets/5354356/6548459/99d10166-c5f9-11e4-972b-523e388c03dd.png)

Basically one instantly sees server/network staff, even if they have no affiliation to the channel.
If this was somehow easy to add, i'd switch over instantly and supply a patch.

But most likely it would make more sense to write a somehow generic nick colorizing system, so people could configure it based on their needs.

**Comment by Zarthus** [comment]
I don't know if network-operator-distinguishment is a priority in particular (but perhaps nice to account for).  I personally don't really like it when people think someone else has higher authority or a higher ranked opinion because of them being a staff, which is often completely unrelated to the channel (and while it may not be the case for you, @somenet, it happens quite often to me, especially where they address network operators in specific because they feel they have the answer to their question, even when it doesn't really matter).

That all said though, I don't mind support for distinguishing network operators, as long as it's not enabled by default.
